### ISSUE TITLE
Delivery guy and Builder, circle logic loop

### ISSUE THREAD

**Issue by roadblockx** [ldtteam/minecolonies#5333]
# Before reporting please make sure that this issue was not reported yet (Check out similar issue suggestions).
## Also update to the latest alpha of both minecolonies and structurize before reporting

Version:  Minecolonies-0.13.3-BETA-universal

### Expected behavior

- I would expect the builder to finish the warehouse and/or the delivery person to make their rounds

- I would also expect the delivery person to make their rounds collecting items from the lumber jack, miner and farmer.

### Actual behavior

The builder continues to walk to the warehouse construction site, stands at the site for several seconds and then walks back to the builder's hut.  He displays the red exclamation mark over his head requesting a delivery person.  Exact message from builder (I need 1 [Deliveryman] to continue working.)  All material he needs for the project is in the builder's hut inventory.

The deliveryman has been found to be sitting in the delivery hut, in the loft of his house and in the town hall.  Not gathering ANY materials.  The red exclamation mark above his head never goes away and when I talk to him, message I get is (Please upgrade or build a warehouse for me to work at)

I can't build the warehouse!  The builder won't complete it because he's looking for a delivery guy and the delivery guy won't cooperate because he wants a warehouse.

To try and resolve this, here are the steps I've performed so far (not necessarily in this order as I've tried so many things with no success)
- restarted MC server, MC client and my laptop (Windows 10 64bit)
- fired the builder, the delivery guy, guard / then set hiring to manual / reassigned all three to new positions
- paused the deliver hut and restarted / then paused the builder's hut and restarted / finally canceled the warehouse construction / started construction again at the same site
- cleared all of their inventories (builder and delivery guy) and the inventories in their huts
- with an empty hand, punched the builder and then the delivery guy (felt good) 
- repaired the delivery guy's hut and the builder's hut (nothing fixed that I could see)
- fired the delivery guy and guard, switched their jobs and restarted their huts
- tried to get them to ignore or skip their request for warehouse or delivery guy 
- recalled everyone to the town hall (shortly after, the delivery guy would not leave the town hall)
- smacked the delivery guy again
- run the requestsystem-reset option
- had enough with the delivery guy and after finding the command to kill him, I sent him to the big MC cloud in the sky
- spawned in a new citizen who I picked as the delivery guy (chick now) / they displayed the exact same error

I don't know if I built the huts out of order which is causing the problems (delivery hut and then warehouse).  If this is the problem, let me know how to fix.

I haven't had any problems prior to this with Minecolonies.  I've built a mine, lumberjack hut, town hall, two citizen huts, tavern, farm and guard tower.  The other mods I have loaded have been there all along and showed no problems (I believe the log file shows the mods installed but if it doesn't, it isn't detailed enough or you need additional info, please ask and I'll do what I can)

Thanks for any help and time you can provide me on this!

### Steps to reproduce the problem

1.  Build warehouse
2.   Notice builder is waiting on delivery guy
3.  Delivery guy is waiting on warehouse
4. ... Profit

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log:   <script src="https://gist.github.com/roadblockx/a2e96f7a1b7f2526bb9fc2c96daca13a.js"></script>
- crashlog:  n/a

**Comment by Raycoms** [comment]
download the newest .14

**Comment by roadblockx** [comment]
Thanks for the quick reply.  Not sure why you closed it out so quickly though as I had a couple questions.  

I’m renting a server for a few family members to play together during quarantine.  I loaded forge for MC 1.15.2 and the other mods all meet that version.  Does .14 work with forge 32.2.x?  That’s the version I had to custom load into the rental server and it was ugly.

**Comment by roadblockx** [comment]
**strike that**

Forge 31.2.28 (trying to fry chicken and remember the version numbers).  My apologies

**Comment by roadblockx** [comment]
Hello?  I’m sure we can do better than a quick post and close the ticket.

**Comment by roadblockx** [comment]
> download the newest .14

This reply did NOT fix the issue.  Future readers, this was closed out before any follow up was done so please don't go through the process of downloading and installing this jar file.  I'll post back what ended up being the fix.

**Comment by Raycoms** [comment]
We close the issue, when the code in question was adjusted in the newest version. That is the way to handle this.
If the issue still persist you reopen the issue and say the issue persists.

**Comment by roadblockx** [comment]
Fair enough as everyone has their own way of supporting their end users.  As a operations engineer for 18 years, I wouldn’t put the burden on the end user to keep creating support tickets for the same issue.  You fix the issue and then close the ticket out as the burden should be on the tech people, not the customer/client/user.  But like I said, everyone has their own way.  Luckily Raven helped and renewed my trust in this mod and group.  It’s a great mod, just a flaky support model.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Raycoms** [comment]
Okay, as a operations engineer for 18 years, what kind of salary did you get, 50-80k a year?

**Comment by Raycoms** [comment]
This is an open source project. Everyone is asked to their part.

**Comment by roadblockx** [comment]
My paid job yes but I volunteer to small business that can’t afford a major IT staff.  Also help veterans for nothing.  Point is?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by roadblockx** [comment]
That’s great.  Open source is a wonderful thing.  Doesn’t change that closing a ticket without knowing whether or not it was the correct diagnosis is a flawed support model.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by someaddons** [comment]
hey if you pay us fulltime then we're happy to improve the support. But as of now this is a complete free time project so its pretty entitled to demand how we spend our freetime which already mainly goes into making a mod for other for free.

**Comment by Raycoms** [comment]
And as I said, we ask the community to do their part. And we do our part of coding and improving this mod.

**Comment by roadblockx** [comment]
Lol.  You don’t like feedback that isn’t positive I take it.  Come on.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Raycoms** [comment]
The point is, as we pointed out, we don't have time for this. That's why we ask the community, that uses this software for free to step up and do their part. Would you rather have great support and no features ever? I rather won't.

**Comment by roadblockx** [comment]
You and your crew are losing sight of the point I was making.  I complimented the mod.  It’s the first one I actually believe should be bought my MS and implemented into MC.  It’s also the only one I’m going to give some money to out of appreciation.  Regardless of all that, my point related to the support model of closing a ticket whether the issue was fixed it or not.  That’s it.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by roadblockx** [comment]
Look, when I offer my free services to small businesses, I don’t make it sound like I should be nailed to a cross.  I’m doing it for the love of technology and helping.  If you aren’t getting the accolades and appreciation you want, maybe don’t do it.  If that’s what you’re after, it’s not worth it for the customer to feel like you’re doing them a major favor.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Raycoms** [comment]
It's because we use issues to track things that need to be solved. If I code something to fix something, and this was merged we close the issue. Meaning for now I can go after other issues. (We do play test things to make sure the fix works). If the issue persists, probably something else is broken and the player creates a new issue and we know someone has to go after that.

**Comment by Raycoms** [comment]
That's how we've been handling this for the past years with our community and it has been proven to be a very efficient model where everyone contributes.

**Comment by roadblockx** [comment]
Good call.  I appreciate everything you and your crew do.  It’s a very fun mod that has allowed countless people to enjoy their time together while being safely in quarantine.  I thank you all for the time and energy you put into it.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by roadblockx** [comment]
Awesome.  Obviously it works very well and doesn’t need to be tweaked.  Continue with it.

I wish you all the best.  Take care!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
